index/scorch/scorch.go

```diff
@@ -38,20 +38,20 @@ const Name = "scorch"
 const Version uint8 = 1
 
 type Scorch struct {
-	readOnly          bool
-	version           uint8
-	config            map[string]interface{}
-	analysisQueue     *index.AnalysisQueue
-	stats             *Stats
-	nextSegmentID     uint64
-	nextSnapshotEpoch uint64
-	path              string
+	readOnly      bool
+	version       uint8
+	config        map[string]interface{}
+	analysisQueue *index.AnalysisQueue
+	stats         *Stats
+	nextSegmentID uint64
+	path          string
 
 	unsafeBatch bool
 
-	rootLock      sync.RWMutex
-	root          *IndexSnapshot // holds 1 ref-count on the root
-	rootPersisted []chan error   // closed when root is persisted
+	rootLock          sync.RWMutex
+	root              *IndexSnapshot // holds 1 ref-count on the root
+	rootPersisted     []chan error   // closed when root is persisted
+	nextSnapshotEpoch uint64
 
```

Comment:
Since eligibleForRemoval and ineligibleForRemoval are also accessed within the rootLock, maybe move them up to this section?